backend/rust-gbt/src/gbt.rs

```diff
@@ -0,0 +1,356 @@
+use priority_queue::PriorityQueue;
+use std::cmp::Ordering;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::f64::INFINITY;
+
+use crate::audit_transaction::AuditTransaction;
+use crate::thread_transaction::ThreadTransaction;
+
+const BLOCK_WEIGHT_UNITS: u32 = 4_000_000;
+const BLOCK_SIGOPS: u32 = 80_000;
+const BLOCK_RESERVED_WEIGHT: u32 = 4_000;
+const MAX_BLOCKS: usize = 8;
+
+struct TxPriority {
+    uid: u32,
+    score: f64,
+}
+impl PartialEq for TxPriority {
+    fn eq(&self, other: &Self) -> bool {
+        self.uid == other.uid
+    }
+}
+impl Eq for TxPriority {}
+impl PartialOrd for TxPriority {
+    fn partial_cmp(&self, other: &TxPriority) -> Option<Ordering> {
+        if self.score == other.score {
+            return Some(self.uid.cmp(&other.uid));
+        } else {
+            return other.score.partial_cmp(&self.score);
+        }
+    }
+}
+impl Ord for TxPriority {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap()
+    }
+}
+
+pub fn gbt(
+    mempool: &mut HashMap<u32, ThreadTransaction>,
+) -> Option<(Vec<Vec<u32>>, Vec<(u32, f64)>, Vec<Vec<u32>>)> {
+    make_block_templates(mempool)
+}
+
+/*
+* Build projected mempool blocks using an approximation of the transaction selection algorithm from Bitcoin Core
+* (see BlockAssembler in https://github.com/bitcoin/bitcoin/blob/master/src/node/miner.cpp)
+* Ported from https://github.com/mempool/mempool/blob/master/backend/src/api/tx-selection-worker.ts
+*/
+fn make_block_templates(
+    mempool: &mut HashMap<u32, ThreadTransaction>,
+) -> Option<(Vec<Vec<u32>>, Vec<(u32, f64)>, Vec<Vec<u32>>)> {
```

Comment:
Why does gbt function exist? Just rename make_block_template to gbt.

```suggestion
/*
* Build projected mempool blocks using an approximation of the transaction selection algorithm from Bitcoin Core
* (see BlockAssembler in https://github.com/bitcoin/bitcoin/blob/master/src/node/miner.cpp)
* Ported from https://github.com/mempool/mempool/blob/master/backend/src/api/tx-selection-worker.ts
*/
pub fn gbt(
    mempool: &mut HashMap<u32, ThreadTransaction>,
) -> Option<(Vec<Vec<u32>>, Vec<(u32, f64)>, Vec<Vec<u32>>)> {
```

Reply:
yup, good point. I think there was some data initialization logic in there originally that evaporated during a refactor